packit_service/worker/handlers/testing_farm.py

```diff
@@ -134,18 +144,18 @@ def _get_or_create_group(
                 type=self.db_trigger.job_trigger_model_type,
                 trigger_id=self.db_trigger.id,
             )
-            if self.testing_farm_job_helper.skip_build or not copr_build
-            else copr_build.group_of_targets.runs[-1]
+            if self.testing_farm_job_helper.skip_build or not builds
+            # All the builds should be in the same copr build group, therefore
+            # connected to the same pipeline, just take the first one
+            else list(builds.values())[0].group_of_targets.runs[-1]
```

Comment:
Do you really need a list if you take just the first one?

Reply:
Oh, `next()` will probably be better, good point